src/core/security/auth.c

```diff
@@ -159,6 +144,7 @@ static void on_host_checked(void *user_data, grpc_security_status status) {
     gpr_asprintf(&error_msg, "Invalid host %s set in :authority metadata.",
                  grpc_mdstr_as_c_string(calld->host));
     bubbleup_error(elem, error_msg);
+    grpc_metadata_batch_destroy(&calld->op.data.metadata);
```

Comment:
Of course.

On Tue, Apr 21, 2015 at 11:59 AM, Craig Tiller <notifications@github.com>
wrote:

> In src/core/security/auth.c
> <https://github.com/grpc/grpc/pull/1312#discussion_r28810880>:
>
> > @@ -159,6 +144,7 @@ static void on_host_checked(void *user_data, grpc_security_status status) {
> >      gpr_asprintf(&error_msg, "Invalid host %s set in :authority metadata.",
> >                   grpc_mdstr_as_c_string(calld->host));
> >      bubbleup_error(elem, error_msg);
> > +    grpc_metadata_batch_destroy(&calld->op.data.metadata);
>
> If I promise to remove the need for it in the next PR in this series, can
> we leave it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/1312/files#r28810880>.
>